src/clickonce/shared/NativeMethods.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Deployment.Utilities
+{
+    internal static class NativeMethods
+    {
+        // Warning - Launcher resource type and name should never be changed
+        public static readonly IntPtr Launcher_CustomResourceTypePtr = new IntPtr(50);
+        public const string Launcher_ResourceName = "FILENAME";
+
+        public const UInt32 LOAD_LIBRARY_AS_DATAFILE = 0x00000002;
+
+        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
```

Comment:
A nice pattern that is more and more common is to name the containing class to be the name of the native binary. e.g.

``` CSharp
internal static class Kernel32
{
     [DllImport(nameof(Kernel32), CharSet = CharSet.Unicode, SetLastError = true)]
     public static extern IntPtr BeginUpdateResourceW(String fileName, bool deleteExistingResource);
}
```

Reply:
That is nice - I'll update it.

Reply:
I intend to keep Kernel32 class inside NativeMethods, as these are still native methods. I could alternatively create NativeMethods namespace and pull Kernel32 out of NativeMerhods. @AaronRobinsonMSFT any preference?

Reply:
>any preference?

Not at all, entirely up to you. I just wanted to let you know a common pattern exists.